generate/schemagen/gen/groups.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package gen
+
+import (
+	reflect "reflect"
+	"strings"
+
+	"github.com/juju/errors"
+	facade "github.com/juju/juju/apiserver/facade"
+	"github.com/juju/rpcreflect"
+)
+
+// FacadeGroup defines the grouping you want to export.
+type FacadeGroup string
+
+const (
+	// Latest gets the latest facades from all the facades.
+	Latest FacadeGroup = "latest"
+	// All gets all the facades no matter the version.
+	All FacadeGroup = "all"
+	// Client facades returns just the client facades along with some required
+	// facades that the client can use.
+	Client FacadeGroup = "client"
+	// JIMM facade group defines a very select set of facades that only work
+	// JIMM.
+	JIMM FacadeGroup = "jimm"
+)
+
+// ParseFacadeGroup will attempt to parse the facade group
+func ParseFacadeGroup(s string) (FacadeGroup, error) {
+	switch s {
+	case "latest", "all", "client", "jimm":
+		return FacadeGroup(s), nil
+	default:
+		return FacadeGroup(""), errors.NotValidf("facade group")
+	}
+}
+
+// Filter the facades based on the group.
+func Filter(g FacadeGroup, facades []facade.Details, registry Registry) []facade.Details {
+	switch g {
+	case Latest:
+		return latestFacades(facades)
+	case All:
+		return allFacades(facades)
+	case Client:
+		return clientFacades(facades, registry)
+	case JIMM:
+		return jimmFacades(facades)
+	}
+	return facades
+}
+
+func latestFacades(facades []facade.Details) []facade.Details {
+	latest := make(map[string]facade.Details)
+	for _, facade := range facades {
+		if f, ok := latest[facade.Name]; ok && facade.Version < f.Version {
+			continue
+		}
+		latest[facade.Name] = facade
+	}
+	latestFacades := make([]facade.Details, 0, len(latest))
+	for _, v := range latest {
+		latestFacades = append(latestFacades, v)
+	}
+	return latestFacades
+}
+
+func allFacades(facades []facade.Details) []facade.Details {
+	return facades
+}
+
+func clientFacades(facades []facade.Details, registry Registry) []facade.Details {
+	required := map[string]struct{}{
+		"AllWatcher":      struct{}{},
+		"AllModelWatcher": struct{}{},
```

Comment:
The client output appears to be missing these facades:
- Admin
- ModelSummaryWatcher
- Pinger

Reply:
Will add